Include/internal/pycore_jit.h

```diff
@@ -11,7 +11,10 @@ extern "C" {
 
 #ifdef _Py_JIT
 
-typedef _Py_CODEUNIT *(*jit_func)(_PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate);
+typedef _Py_CODEUNIT *(*jit_func_native)(_PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate);
+
+// To use preserve_none in JIT builds, we need to declare a separate function pointer with __attribute__((preserve_none)) since this attribute is not supported in < clang 19.
+typedef _Py_CODEUNIT *(*jit_func_preserve_none)(_PyInterpreterFrame *frame, _PyStackRef *stack_pointer, PyThreadState *tstate) __attribute__((preserve_none));
```

Comment:
This should be moved somewhere that only `Tools/jit/template.c` and `Tools/jit/trampoline.c` can see it. Either just copy it to each file, or add a new header in `Tools/jit` that's used by both. Otherwise the "normal" C compiler complains about the new attribute when building the rest of the interpreter.